<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <!--arguments-->
    <arg name="camera_ID_1" default="/dev/video0"/>
    <arg name="camera_name_1" default="camera_1"/>
    <arg name="subscriber_name_1" default="subscriber_1"/>

    <arg name="camera_ID_2" default="/dev/video2"/>
    <arg name="camera_name_2" default="camera_2"/>
    <arg name="subscriber_name_2" default="subscriber_2"/>

    <arg name="subscriber_name_3" default="subscriber_3"/>

    <arg name="fps" default = "30"/>


    <!-- Argument for the first camera -->
    <!-- <arg name="camera_name_1" default="/dev/video0"/> -->
    <!-- Argument for the second camera -->
    <!-- <arg name="camera_name_2" default="/dev/video2"/> -->

    <!-- Node for the first camera -->
    <node name="$(arg camera_name_1)" pkg="testpackage" type="image_publisher_launch.py" output="screen" args="$(arg camera_ID_1) /image_raw_1 $(arg fps)">
        <param name="camera_name" type="string" value="$(arg camera_name_1)" />
    </node>

    <!-- Node for the second camera -->
    <node name="$(arg camera_name_2)" pkg="testpackage" type="image_publisher_launch.py" output="screen" args="$(arg camera_ID_2) /image_raw_2 $(arg fps)">
        <param name="camera_name" type="string" value="$(arg camera_name_2)" />
    </node>

    <!-- Subscriber node for the first camera -->
    <node name="$(arg subscriber_name_1)" pkg="testpackage" type="image_subscriber.py" args="/image_raw_1" output="screen">
        <param name="topic_name" type="string" value="/image_raw_1" />
    </node>

    <!-- Subscriber node for the second camera -->
    <node name="$(arg subscriber_name_2)" pkg="testpackage" type="image_subscriber.py" args="/image_raw_2" output="screen">
        <param name="topic_name" type="string" value="/image_raw_2" />
    </node>

    <!-- yolov5 detection node-->
    <node name="yolov5_node" pkg="testpackage" type="yolov5_node.py" output="screen" args="/image_raw_1 /yolov5/detections" />

    <!-- Subscriber node for the detection node -->
    <node name="$(arg subscriber_name_3)" pkg="testpackage" type="image_subscriber.py" args="/yolov5/detections" output="screen">
        <param name="topic_name" type="string" value="/image_raw_2" />
    </node>
</launch>